src/hasher.cpp

```diff
@@ -77,14 +49,7 @@ namespace libtorrent
 #elif TORRENT_USE_COMMONCRYPTO
 		CC_SHA1_Init(&m_context);
 #elif TORRENT_USE_CRYPTOAPI
-		if (CryptCreateHash(get_crypt_provider(), CALG_SHA1, 0, 0, &m_context) == false)
-		{
-#ifndef BOOST_NO_EXCEPTIONS
-			throw system_error(error_code(GetLastError(), system_category()));
-#else
-			std::terminate();
-#endif
-		}
+		m_context.create();
```

Comment:
if you make ``create``, ``destroy`` and ``duplicate`` the constructor, destructor and copy constructor/copy assignment, you don't need to make these calls.